snapcraft/plugins/catkin.py

```diff
@@ -372,8 +383,21 @@ def pull(self):
             if self.options.source_subdir:
                 source_path = os.path.join(self.sourcedir,
                                            self.options.source_subdir)
-            _handle_rosinstall_files(
-                wstool,  source_path, self.options.rosinstall_files)
+
+            # Create a set of paths to rosinstall files
```

Comment:
this seems to only be used in your `else` clause bellow, why not just define it there? I also do not understand why you just don't pass `self.options.rosinstall_files` to `_handle_rosinstall_files`

Reply:
A `source_path`

Reply:
``python
else:
    rosinstall_files_full_paths = {os.path.join(source_path, ri) for ri in self.options.rosinstall_files} 
    _handler_rosinstall_files(wstool, rosinstall_files_full_paths)
```

maybe?

Reply:
Good catch, it should be under the `else` indeed. As to why we don't just pass the option itself, it's because in this PR the `_handle_rosinstall_files` function was updated to no longer accept the parent directory path, and just take absolute paths to rosinstall files instead. This was done to make the recursive case easier and more efficient. Otherwise we'd need to process each rosinstall path recursively discovered, figure out the common parent directory, then pass that and relative paths onto `_handle_rosinstall_files`.